package/network/services/wireguard/Makefile

```diff
@@ -25,64 +25,54 @@ PKG_BUILD_DIR:=$(KERNEL_BUILD_DIR)/wireguard-linux-compat-$(PKG_VERSION)
 PKG_BUILD_PARALLEL:=1
 PKG_USE_MIPS16:=0
 
+PKG_MAINTAINER:=Jason A. Donenfeld <Jason@zx2c4.com>
+
 # WireGuard's makefile needs this to know where to build the kernel module
 export KERNELDIR:=$(LINUX_DIR)
 
 include $(INCLUDE_DIR)/package.mk
 
-define Package/wireguard/Default
-  SECTION:=net
-  CATEGORY:=Network
-  SUBMENU:=VPN
+define KernelPackage/wireguard/Defaults
+  SECTION:=kernel
+  CATEGORY:=Kernel modules
+  SUBMENU:=Network Support
+  TITLE:=WireGuard kernel module
   URL:=https://www.wireguard.com
-  MAINTAINER:=Jason A. Donenfeld <Jason@zx2c4.com>
 endef
 
-define Package/wireguard/Default/description
+define KernelPackage/wireguard
+  $(call KernelPackage/wireguard/Defaults)
+  DEPENDS:=+!LINUX_5_4:kmod-wireguard-it +LINUX_5_4:kmod-wireguard-oot
```

Comment:
Won't it be better to do this in wireguard-tools?

Reply:
No - we do not want kernel symbol dependencies in user space. See https://github.com/openwrt/packages/issues/12072 and https://github.com/openwrt/openwrt/pull/3039

Reply:
Okay, thanks for explanation.